bundles/org.openhab.binding.rotel/src/main/java/org/openhab/binding/rotel/internal/handler/RotelHandler.java

```diff
@@ -1488,9 +1549,23 @@ public void onNewMessageEvent(EventObject event) {
                     }
                     break;
                 case KEY_SOURCE:
-                    sources[0] = model.getSourceFromCommand(RotelCommand.getFromAsciiCommand(value));
+                    source = model.getSourceFromCommand(RotelCommand.getFromAsciiCommand(value));
+                    sources[0] = source;
                     updateChannelState(CHANNEL_SOURCE);
                     updateChannelState(CHANNEL_MAIN_SOURCE);
+                    RotelCommand presetGetCmd = getRadioPresetGetCommand(source);
+                    if (presetGetCmd != null) {
+                        // Request current preset (with a delay)
+                        scheduler.schedule(() -> {
+                            try {
+                                sendCommand(presetGetCmd);
+                            } catch (RotelException e) {
+                            }
```

Comment:
Consider adding a comment here why this exception is ignored.

Reply:
If I remove the try/catch, I have an error about unhandled exception.
So I added a log.